src/domain/usecases/MapDiseaseOutbreakToAlertsUseCase.ts

```diff
@@ -0,0 +1,48 @@
+import { FutureData } from "../../data/api-futures";
+import { AlertRepository } from "../repositories/AlertRepository";
+import _ from "../entities/generic/Collection";
+import { Id } from "../entities/Ref";
+import { DiseaseOutbreakEventBaseAttrs } from "../entities/disease-outbreak-event/DiseaseOutbreakEvent";
+import { Future } from "../entities/generic/Future";
+import {
+    hazardTypeCodeMap,
+    incidentStatusTypeMap,
+    RTSL_ZEBRA_ALERTS_DISEASE_TEA_ID,
+    RTSL_ZEBRA_ALERTS_EVENT_TYPE_TEA_ID,
+} from "../../data/repositories/consts/DiseaseOutbreakConstants";
+
+type DiseaseOutbreakEventData = Pick<
+    DiseaseOutbreakEventBaseAttrs,
+    "dataSource" | "hazardType" | "incidentStatus" | "suspectedDiseaseCode"
+>;
+
+const incidentDataSourceCode = "IBS";
+
+export class MapDiseaseOutbreakToAlertsUseCase {
+    constructor(private alertRepository: AlertRepository) {}
+
+    public execute(
+        diseaseOutbreakEventId: Id,
+        diseaseOutbreakEventData: DiseaseOutbreakEventData
+    ): FutureData<void> {
+        const { dataSource, hazardType, incidentStatus, suspectedDiseaseCode } =
+            diseaseOutbreakEventData;
+
+        const hazardTypeCode = hazardType ? hazardTypeCodeMap[hazardType] : undefined;
+        const incidentStatusTypeCode = incidentStatusTypeMap[incidentStatus];
+
+        if (!diseaseOutbreakEventId)
+            return Future.error(new Error("Disease Outbreak Event Id is required"));
+
+        const filter =
```

Comment:
Create this filter in the data repository because it's related to DHIS2 (remember the inner layers don't have to know about the outer layers)